src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/integration/CardValidatorTest.java

```diff
@@ -0,0 +1,272 @@
+            .withHandForPlayer(player.getName())
+            .withCards(ACE_OF_CLUBS, EIGHT_OF_CLUBS)
+            .withAnnouncedScore(1)
+            .finishHand()
+            .withHandForPlayer(PLAYER_NAME_2)
+            .withCards(JACK_OF_SPADES, QUEEN_OF_HEARTS)
+            .withAnnouncedScore(0)
+            .finishHand()
+            .withRound()
+            .finishRound()
+            .finishMatch()
+            .build();
+
+    gameRepository.saveAll(List.of(game));
+    Hand hand =
+        game.getLastMatch().orElseThrow().getHands().stream()
+            .filter(
+                h ->
+                    h.getCards().stream().filter(c -> c.getCardRank() == CardRank.EIGHT).count()
+                        > 0)
+            .findAny()
+            .orElseThrow();
+    Card card =
+        hand.getCards().stream()
+            .filter(c -> c.getCardRank() == CardRank.EIGHT)
+            .findAny()
+            .orElseThrow();
+    Round round = game.getLastMatch().orElseThrow().getLastRound().orElseThrow();
+    CARD_BEFORE.setRound(round);
+    CARD_BEFORE.setHand(hand);
+    card.setId(1L);
+    card.setRound(round);
+    Exception exception =
+        assertThrows(HttpClientErrorException.class, () -> cardValidator.onUpdateCard(card));
+    assertEquals("422 It's not your turn.", exception.getMessage());
+  }
+
+  @Test
+  @WithPersistedPlayer(playerName = PLAYER_NAME_1)
+  void playing_card_in_announcing_round_throws_error() {
+    Player player = playerRepository.findByName(PLAYER_NAME_1).orElseThrow();
+    game =
+        GameBuilder.builder("game1", gameRepository, participationRepository, playerRepository)
+            .withParticipationWith(player)
+            .withParticipation(PLAYER_NAME_2)
+            .withGameState(GameState.PLAYING)
+            .withMatch()
+            .withMatchState(MatchState.ANNOUNCING)
+            .withHandForPlayer(player.getName())
+            .withCards(ACE_OF_CLUBS, EIGHT_OF_CLUBS)
+            .finishHand()
+            .withHandForPlayer(PLAYER_NAME_2)
+            .withCards(JACK_OF_SPADES, QUEEN_OF_HEARTS)
+            .finishHand()
+            .withRound()
+            .finishRound()
+            .finishMatch()
+            .build();
+
+    gameRepository.saveAll(List.of(game));
+    Card card =
+        game.getLastMatch().orElseThrow().getSortedHands().stream()
+            .findAny()
+            .orElseThrow()
+            .getCards()
+            .stream()
+            .filter(c -> c.getCardRank() == CardRank.EIGHT)
+            .findAny()
+            .orElseThrow();
+    Round round = game.getLastMatch().orElseThrow().getLastRound().orElseThrow();
+    card.setRound(round);
+    Exception exception =
+        assertThrows(HttpClientErrorException.class, () -> cardValidator.onUpdateCard(card));
+    assertEquals(
+        "422 You can not play a card before everybody has announced the score.",
+        exception.getMessage());
+  }
+}
```

Comment:
Wow...line hunter? ;-)

Reply:
Ich has glaubs verstande du fräche Hagel :angry: :joy: No, my dear friend, but the error code is always UNPROCESSABLE ENTITY and I want to ensure that I really get into the expected if and not in any. I can only verify that by checkint the specific exception message.